enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,333 @@
+3. The new container process gathers Insights data based on the configuration provided in the CR instance and is responsible for uploading the gathered Insights archive and checking that the archive was processed correctly in the console.redhat.com (the API documentation is available at https://console.redhat.com/docs/api/insights-results-aggregator/v2). 
+4. The original Insights operator's pod is responsible for downloading the last known Insights analysis report (from console.redhat.com)
+   based on latest successfully finished gathering job and updating the corresponding values in the already existing `insightsoperator.operator.openshift.io` resource.
+
+This scenario brings few questions and things to be discussed. Please see the following sections [Implementation details](#implementation-detailsnotesconstraints), [Risk mitigations](#risks-and-mitigations) and [Open questions](#open-questions)
+
+### API Extensions
+
+This is adding new Insights CRD in the `insights.openshift.io/v1alpha1` . The proposal of the structure of the new CRD is mentioned in the [Proposal](#Proposal) section.
+
+### Implementation Details/Notes/Constraints
+
+#### Current behavior
+The Insights operator currently runs and gathers the data in a single container. 
+Uploading of new archives, pruning old archives as well as downloading the last Insights analysis report (this is always per archive) is implemented in this one container/binary. 
+When there is a successful upload (to the console.redhat.com) of the archive then there is some short delay (waiting for the processing in console.redhat.com pipelines - see the [API documentation](https://console.redhat.com/docs/api/insights-results-aggregator/v2)) and then there is a request to get the last Insights analysis report per given cluster. 
+Finally the status section of the `insightsoperator.operator.openshift.io` CR is updated. Note that there is a retry logic in case of failed upload and after some time (and number of attempts) the Insights operator is marked as degraded.
+
+#### Updates required to allow on-demand data gathering
+The plan is to refactor the existing periodic data gathering to run in a separate Pod/Job. The existing Insights operator pod will still exist, but it would periodically do the following:
+- check the `insightsdatagather.config.openshift.io` CR to read current user configuration
+- create a new `datagather.insights.openshift.io` CR (in the `openshift-insights` namespace) based on the values from the above
+- spawn a new `job` to gather the Insights data, upload it to `console.redhat.com` and check that the archive was processed in console.redhat.com (this requires a new API endpoint to be implemented in https://console.redhat.com/docs/api/insights-results-aggregator/v2)
+- status attribute of the `datagather.insights.openshift.io` is updated within the job execution
```

Comment:
Which attribute is being updated here? I don't see an appropriate field to hold a reference to the uploaded artefact?

Reply:
Yeah good point. Let me add the new attribute.

Reply:
Updated

Reply:
Is this up to date, is the attribute the request id field you've recently added?